tsl/src/bgw_policy/job_api.c

```diff
@@ -87,8 +143,12 @@ job_add(PG_FUNCTION_ARGS)
 	namestrcpy(&proc_name, func_name);
 	namestrcpy(&owner_name, GetUserNameFromId(owner, false));
 
+	/* The check exists but may not have the expected signature: (config jsonb) */
+	if (OidIsValid(check))
+		validate_check_signature(check);
+
 	if (config)
-		job_config_check(&proc_schema, &proc_name, config);
+		ts_bgw_job_run_config_check(check, 0, config);
```

Comment:
We should probably always run the config check function if it is defined, even if the config is NULL. It should be up to the check function to decide if NULL config is acceptable since providing no configuration might be a failure case. For instance, some of the tests for custom jobs check for "drop_after" and fails if not present in the config. Passing no config at all is the same as passing a JSON without "drop_after".

Reply:
Agreed, will fix